ls1intum/Artemis
# `General`: Fix an issue when hovering over tooltips

### Checklist
#### General


#### Client

### Motivation and Context


When creating a tooltip by hovering over an element, the tooltip is put in the same container as the anchor element. If the tooltip now overflows the previous container size, the container gets resized, which leads to a shifting of other elements on the page when a tooltip appears or vanishes.

### Description

I changed the default value for tooltips, so they get attached to the 'body' element. This way we do not have to change every existing tooltip and it also makes it easier for tooltips created in the future, as the developer does not have to know this possibly unwanted behavior.

NOTE: Even when we attach the tooltips to the 'body' element, overflows can still happen by bad use of the `placement `attribute. For example, if an element is placed on the right side of the screen and we define `placement='right'`. Then we get some overflow to the right, when the popup appears and ugly horizontal scrolling.
We can prevent this by defining fallback options, as bootstrap uses the first value, that doesn't cause overflow. There is also an `auto` option, that tries to put the element to the top, bottom, left, and right. In that order. So `placement='right auto'` will do the same as `placement='right'` in case there is enough space, but otherwise will put the tooltip to the top, bottom or left instead of forcing an overflow.
I already added the fallback option `auto` to every use of `placement`, but we have to keep that in mind for future use.

### Steps for Testing





Prerequisites:
- none

1. Log in to Artemis
2. Hover over an element, that will spawn a tooltip and verify, that other elements on the page do not get shifted

### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage





### Screenshots


Before Change:

https://user-images.githubusercontent.com/44734881/201890596-e048a7bc-e46c-4dfb-af87-baa285df362f.mov

After Change:

https://user-images.githubusercontent.com/44734881/201891607-fead39e6-045a-49b2-85f3-576c5cd50446.mov
